src/scripts.js

```diff
@@ -96,15 +125,34 @@ function cardButtonConditionals(event) {
   if (event.target.classList.contains('favorite')) {
     favoriteCard(event);
   }
-  if (event.target.classList.contains('card-picture')) {
+  if (event.target.classList.contains('instruction')) {
+    declareRecipe(event)
     findDirections(event);
   } 
-  if(event.target.classList.contains('add-button')){
-    // console.log(typeof event.target.id)
+  if(event.target.classList.contains('add-button')) {
+    console.log(event)
     addToCookLater(event);
   }
- }
-
+  if(event.target.classList.contains('icon')) {
+    let id = event.target.id
+    let currentRecipe = cookbook.recipes.find(x => x.id === +id)
+    let ingredientFeedback = Object.values(userPantry.giveFeedbackOnIngredients(currentRecipe, cookbook))
+    domUpdates.displayIngredientFeedback(ingredientFeedback, id)
+  }
+  if(event.target.classList.contains('back-button')){
+    domUpdates.populateCards(cookbook.recipes, cardArea, currentFavs())
+  }
+  if(event.target.classList.contains('purchase-button')){
+    let id = event.target.id
+    const recipe = declareRecipe(event)
+    const post = postIngredients(recipe) 
+    post.then(values => Promise.all(values))
+    .then(data => data.map(response => response.json()))
+    .then(promises => Promise.all(promises))
+    .then(messages => domUpdates.updatePurchase(messages, id))
```

Comment:
this was painful but maid it trough! 👍 
One thing a leaned today was the .json() also returns a promise and had to parse it back into data we could use. Once I figured it out it was just a matter to link it to the Dom.

Reply:
Weird, right!? I was stuck on an issues stemming from that for so long. The web APIs are really tricky and are teaching me to be way more cautious about how I handle data flow.